aten/src/ATen/native/native_functions.yaml

```diff
@@ -6514,27 +6512,23 @@
 
 - func: fmod.Scalar_out(Tensor self, Scalar other, *, Tensor(a!) out) -> Tensor(a!)
   dispatch:
-    CPU: fmod_out
-    CUDA: fmod_cuda_out
+    CPU, CUDA: fmod_out
 
 - func: fmod.Scalar(Tensor self, Scalar other) -> Tensor
   use_c10_dispatcher: full
   variants: method, function
   dispatch:
-    CPU: fmod
-    CUDA: fmod_cuda
+    CPU, CUDA: fmod
 
 - func: fmod.Tensor_out(Tensor self, Tensor other, *, Tensor(a!) out) -> Tensor(a!)
   dispatch:
-    CPU: fmod_out
-    CUDA: fmod_cuda_out
+    CPU, CUDA: fmod_out
```

Comment:
Now that we're using fmod_out for the CUDA version of out... my question is, are the out= semantics the same as before?

In TH, we had "resize semantics" -- the tensor passed to out= gets resized to be the correct output shape, no matter what the input shape is. For example, 
```
# Before this PR; I checked this on PyTorch 1.6
x = torch.randn(2, 3, 4, 5, device='cuda')
y = torch.randn(10, device='cuda')
z = torch.randn(10, device='cuda')
torch.fmod(y, z, out=x)
assert x.shape == [10]
```

I think we haven't changed the out= semantics, but TensorIterator does have several options for picking various out= semantics. Could we add a test to confirm the out= semantics haven't changed?